index.js

```diff
@@ -23,7 +23,7 @@ class Option {
     this.required = flags.indexOf('<') >= 0; // A value must be supplied when the option is specified.
     this.optional = flags.indexOf('[') >= 0; // A value is optional when the option is specified.
     this.mandatory = false; // The option must have a value after parsing, which usually means it must be specified on command line.
-    this.negate = flags.indexOf('-no-') !== -1;
+    this.negate = flags.indexOf('--no-') !== -1;
```

Comment:
This should test for it being at the start of the string. The other places in the code use a regular expression, but checking the index would be fine here I think.

Reply:
How about something like this?

```suggestion
    this.negate = flags.startsWith('--no-');
```

Reply:
Even better! (We moved away from using index to using routines in #1264 which is on the develop branch.)